include/dali/core/mm/cuda_vm_resource.h

```diff
@@ -212,9 +212,7 @@ class cuda_vm_resource_base : public memory_resource<memory_kind::device> {
     try {
       map_storage(va, size);
     } catch (std::bad_alloc &) {
-      print(std::cerr, "Could not allocate physical storage of size ", size, " on device ",
-        device_ordinal_);
-      dbg_dump(std::cerr);
+      free_va_.put(va, size);
```

Comment:
I think we still should provide a message like - `Attempted to allocate XXX, currently allocated YY, available ZZZ`

Reply:
I don't want to change the original exception type - now I'd have to rethrow.

Reply:
Maybe you can adjust at least adjust `dbg_dump(std::cerr);` to print shorter and easier to digest message?

Reply:
Done for CUDABadAlloc - others are rethrown directly.